src/lib.rs

```diff
@@ -82,38 +104,72 @@ pub fn init(interface: WlcInterface) -> Option<fn() -> ()> {
+/// the `callbacks` module, and the logger using `log_set_handler` or
+/// `log_set_default_handler`. Then call `init2()`.
+///
+/// # Permissions
+/// If a compositor is initialized from the tty using suid or logind, it will
+/// drop extra permissions after a call to `init2()`. It is strongly
+/// recommended to delay code which is not registering callbacks until after
+/// this call.
+///
+/// # wlc Example
+/// ```no_run
+/// use rustwlc;
+/// use rustwlc::callback;
+/// use rustwlc::handle::WlcView;
+///
+/// // An example callback function
+/// // See the various functions in this module for more information
+/// extern "C" fn view_focus_callback(view: WlcView, focused: bool) {
+///     println!("A view came into focus!");
+/// }
+///
+/// // Set a default log callback
+/// rustwlc::log_set_default_handler();
+///
+/// // Register some callbacks
+/// callback::view_focus(view_focus_callback);
+/// // ... and additional callbacks
+///
+/// // The only thing your code should do before init2 is register callbacks
+/// // and log handlers.
+/// let run_wlc = rustwlc::init2()
+///     .expect("Unable to initialize wlc!");
+///
+/// run_wlc();
+/// ```
+pub fn init2() -> Option<fn() -> ()> {
+    unsafe {
+        if wlc_init2() {
+            Some(run_wlc)
+        }
+        else {
+            None
+        }
+    }
+}
+
 /// Runs wlc's event loop.
 ///
 /// The initialize functions will return this function in an Option.
-/// If and only if they succeed can this function be called wlc with `rustwlc::init` call this method
-/// to being wlc's main event loop.
+/// Only then can it be called to being wlc's main event loop.
 fn run_wlc() {
     unsafe {
         wlc_run();
     }
 }
 
-/// Deprecated, do not use.
+/// Deprecated attempt at wlc_exec wrapper.
 ///
 /// # Deprecated
-/// This function does not seem to work across the FFI boundary, and Rust provides a
-/// much better interface in the `std::command::Command` class to executing programs.
-pub fn exec(bin: String, args: Vec<String>) {
-    unsafe {
-        let bin_c = CString::new(bin).unwrap().as_ptr() as *const libc::c_char;
-
-        let argv: Vec<CString> = args.into_iter()
-                                     .map(|arg| CString::new(arg).unwrap())
-                                     .collect();
-
-        let args: Vec<*const libc::c_char> = argv.into_iter()
-                                                 .map(|arg: CString| {
-                                                     arg.as_ptr() as *const libc::c_char
-                                                 })
-                                                 .collect();
-
-        wlc_exec(bin_c, args.as_ptr() as *const *const libc::c_char);
-    }
+/// This function does not seem to work across the FFI boundary, and Rust
+/// provides a much better interface in the `std::command::Command` class to
+// execute programs.
+pub fn exec(_bin: String, _args: Vec<String>) {
```

Comment:
Then why is it still here and doing nothing? If we have a something to replace it with then keep it, if you want to remove it then remove it. If neither, then at least deprecate it if we aren't going to use it anymore